### PR TITLE
fix: (pools) Non staked Cake vault shown when staked only selected

### LINKED ISSUE #1042: [BUG] Syrup pool : staked only / new pool auto CAKE
In Syrup Pools, check staked only
--> the new pool auto CAKE is displayed even if im not in.

### PR DESCRIPTION
**Pull request by memoyil**
Fixes #1042

### CODE REVIEW

File: src/views/Pools/components/CakeVaultCard/index.tsx [hunk 1]
```diff
@@ -35,6 +39,10 @@ const CakeVaultCard: React.FC<{ pool: Pool; account: string }> = ({ pool, accoun
   const stakingTokenPrice = useGetApiPrice(stakingToken.address ? getAddress(stakingToken.address) : '')
   const isLoading = !pool.userData || !userInfo.shares
 
+  if (showStakedOnly && !accountHasSharesStaked) {
+    return <></>
```
Comment (Chef-Cheems) [inline src/views/Pools/components/CakeVaultCard/index.tsx:43]: I think `return null` is more common way to render nothing

Comment (memoyil) [reply src/views/Pools/components/CakeVaultCard/index.tsx]: Changed to null

File: src/views/Pools/components/CakeVaultCard/index.tsx [hunk 2]
```diff
@@ -22,7 +22,11 @@ const StyledCardBody = styled(CardBody)<{ isLoading: boolean }>`
   min-height: ${({ isLoading }) => (isLoading ? '0' : '254px')};
 `
 
-const CakeVaultCard: React.FC<{ pool: Pool; account: string }> = ({ pool, account }) => {
+const CakeVaultCard: React.FC<{ pool: Pool; account: string; showStakedOnly: boolean }> = ({
```
Comment (ChefHutch) [inline src/views/Pools/components/CakeVaultCard/index.tsx:25]: Let's turn it into an interface as there are 3 props now?

**Review by ChefHutch — APPROVED** [review]
Not sure there is a perfectly neat solution to this, but this LGTM.

`The CakeVaultCard` will still make its smart contract calls via the `useGetVaultUserInfo`, `useGetVaultFees` & `useGetVaultSharesInfo` even though nothing is rendered to the DOM, but the alternative (making the `useGetVaultUserInfo` call within the pools view, declaring the `accountHasSharesStaked` boolean and then passing these down) isn't exactly neat.

**Comment by memoyil** [comment]
> Not sure there is a perfectly neat solution to this, but this LGTM.
> 
> The `CakeVaultCard` will still make its smart contract calls via the `useGetVaultUserInfo`, `useGetVaultFees` & `useGetVaultSharesInfo` even though nothing is rendered to the DOM, but the alternative (making the `useGetVaultUserInfo` call within the pools view, declaring the `accountHasSharesStaked` boolean and then passing these down) isn't exactly neat.

Yes either way we have to call, thats why i selected workaroundish way

File: src/views/Pools/components/CakeVaultCard/index.tsx (hunk 2 above)
Comment (memoyil) [reply src/views/Pools/components/CakeVaultCard/index.tsx]: done

**Review by ChefHutch — APPROVED** [review]

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**